### PR TITLE
fix: Switch WebUtility with Uri.EscapeDataString space to %20

### PR DESCRIPTION
**Pull request by DvirCyberArk**
**What does this pull request do?**
Fix space issue while loading policy which contains space at URL

**What background context can you provide?**
Until now we used `WebUtility.UrlEncode(" ")` which yields `"+"` instead of `%20` resulting in Conjur Ngnix error and trimming of URL.

**Where should the reviewer start?**
It seems like many classes to review, however the critical change was at the C'tor at path member. Notice that not all the URL get encoded, just the input variables. Therefore there will be no change to other special char like "?" or "&".

Look at the variableTest for a good example.

**How should this be manually tested?**
Create simple policy with: policy, variable resources that contain spaces. e.g:
```
- !policy
   id: demo vault

- !variable
   id: demo variable
```

and load it using V5 .NET SDK and compare it to same policy loaded from CLI viewing result in UI
and Conjur Ngnix log.

**Link to build in Jenkins (if appropriate)**
https://jenkins.conjur.net/view/Conjur%205.x/job/cyberark--conjur-api-dotnet/

**Questions:**
 - Why using `Uri.EscapeDataStrin` instead of `EscapeUriString`?
https://stackoverflow.com/questions/4396598/whats-the-difference-between-escapeuristring-and-escapedatastring
https://stackoverflow.com/questions/602642/server-urlencode-vs-httputility-urlencode/47877559#47877559
TL;DR:
There is no valid reason to ever use Uri.EscapeUriString. If you need to percent-encode a string, always use Uri.EscapeDataString.

### CODE REVIEW

File: test/VariablesTest.cs [hunk 1]
```diff
@@ -22,8 +22,8 @@ public void GetVariableTest()
             Assert.AreEqual("testvalue", Client.Variable("foo/bar").GetValue());
 
             // TODO: not sure if this is supposed to be a plus or %20 or either
-            Mocker.Mock(new Uri("test:///secrets/" + TestAccount + "/variable/foo+bar"), "space test");
-            Assert.AreEqual("space test", Client.Variable("foo bar").GetValue());
+            // since we are using EscapeDataString %20 is convert to space, however plus is not converted anymore,
+            Mocker.Mock (new Uri ("test:///secrets/" + TestAccount + "/variable/foo%20bar"), "space test");
```
Comment (jvanderhoof) [inline test/VariablesTest.cs]: What is this test doing?  It looks like you're just mocking the response to return a variable with a space in it.  It looks like your changes enable variable names with spaces to be correctly handled.

Comment (DvirCyberArk) [reply test/VariablesTest.cs]: This test was updated - Upon creating a variable with space, a URL (of variable path member) was created. Before update, spaces was converted to + and after update, I verify that they are converted to %20
